qdrant/deployment/src/main/java/io/quarkiverse/langchain4j/qdrant/QdrantBuildConfig.java

```diff
@@ -0,0 +1,78 @@
+package io.quarkiverse.langchain4j.qdrant;
+
+import static io.quarkus.runtime.annotations.ConfigPhase.BUILD_TIME;
+
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+
+@ConfigRoot(phase = BUILD_TIME)
+@ConfigMapping(prefix = "quarkus.langchain4j.qdrant")
+public interface QdrantBuildConfig {
+    /**
+     * Configuration for DevServices. DevServices allows Quarkus to automatically start a database in dev and test mode.
+     */
+    DevServicesConfig devservices();
+
+    @ConfigGroup
+    interface DevServicesConfig {
+
+        /**
+         * Whether Dev Services for Qdrant are enabled or not.
+         */
+        @WithDefault("true")
+        boolean enabled();
+
+        /**
+         * Container image for Qdrant.
+         */
+        @WithDefault("docker.io/qdrant/qdrant:v1.8.3-unprivileged")
+        String qdrantImageName();
+
+        /**
+         * Optional fixed port the Qdrant dev service will listen to.
+         * If not defined, the port will be chosen randomly.
+         */
+        OptionalInt port();
+
+        /**
+         * Indicates if the Dev Service containers managed by Quarkus for Qdrant are shared.
+         */
+        @WithDefault("true")
+        boolean shared();
+
+        /**
+         * Service label to apply to created Dev Services containers.
+         */
+        @WithDefault("qdrant")
+        String serviceName();
+
+        /**
+         * The Qdrant collection configuration.
+         */
+        Optional<CollectionConfig> collection();
+
+        interface CollectionConfig {
+            /**
+             * The vector parameters.
+             */
+            VectorParamsConfig vectorParams();
+        }
+
+        interface VectorParamsConfig {
+            /**
+             * Distance function used for comparing vectors
+             */
+            io.qdrant.client.grpc.Collections.Distance distance();
```

Comment:
I'm not completely sure it's appropriate for us to use an enum from upstream code in our config
- If we migrate away towards our own client (as we discussed), this might need a breaking change
- If the upstream code introduces changes, this might mean a breaking change on our side too
- If they add any funky stuff in the enum, it might break bytecode recording

I'd say we could rather create a copy of this. @geoand WDYT?

Reply:
I'm off today, so I'll have to think about this tomorrow

Reply:
would be ok to define a wrapper enum ? like 

```java
enum Distance {
    Cosine(io.qdrant.client.grpc.Collections.Distance.Cosine);
    
    Distance(io.qdrant.client.grpc.Collections.Distance) {
     ...
    }
}
```

Reply:
That would still depend on the upstream class :( 
I'd say we should have a copy with all possible (string-based) values and some kind of adapter that maps from our enum  to the upstream enum

Reply:
done